doc/source/cluster/kubernetes/examples/gpu-training-example.md

```diff
@@ -0,0 +1,150 @@
+(kuberay-gpu-training-example)=
+
+# ML training with GPUs on Kubernetes
+In this guide, we show you how to run a sample Ray machine learning training workload with GPU on Kubernetes infrastructure. We will run Ray's {ref}`PyTorch image training benchmark <pytorch_gpu_training_benchmark>` with a 1 gigabyte training set.
+
+:::{note}
+To learn the basics of Ray on Kubernetes, we recommend taking a look
+at the {ref}`introductory guide <kuberay-quickstart>` first.
+:::
+
+## The end-to-end workflow
+The following script summarizes the end-to-end workflow for GPU training. These instructions are for GCP, but a similar setup would work for any major cloud provider. The following script consists of:
+- Step 1: Set up a Kubernetes cluster on GCP.
+- Step 2: Deploy a Ray cluster on Kubernetes with the KubeRay operator.
+- Step 3: Run the PyTorch image training benchmark. 
+
+```shell
+# Step 1: Set up a Kubernetes cluster on GCP
+# Create a node-pool for a CPU-only head node
+# e2-standard-8 => 8 vCPU; 32 GB RAM
+gcloud container clusters create gpu-cluster-1 \
+    --num-nodes=1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+    --zone=us-central1-c --machine-type e2-standard-8
+
+# Create a node-pool for GPU. The node is for a GPU Ray worker node.
+# n1-standard-8 => 8 vCPU; 30 GB RAM
+gcloud container node-pools create gpu-node-pool \
+  --accelerator type=nvidia-tesla-t4,count=1 \
+  --zone us-central1-c --cluster gpu-cluster-1 \
+  --num-nodes 1 --min-nodes 0 --max-nodes 1 --enable-autoscaling \
+  --machine-type n1-standard-8
+
+# Install NVIDIA GPU device driver
+kubectl apply -f https://raw.githubusercontent.com/GoogleCloudPlatform/container-engine-accelerators/master/nvidia-driver-installer/cos/daemonset-preloaded.yaml
+
+# Step 2: Deploy a Ray cluster on Kubernetes with the KubeRay operator.
+# Please make sure you are connected to your Kubernetes cluster. (For GCP, you can do so by running the relevant "gcloud containers" command.)
```

Comment:
What is the relevant `gcloud containers` command (same below) -- there is only `gcloud container` I think and I wonder which subcommand should be used :)

Reply:
Right, doesn't hurt to give more details here.
Let's spell out the command:
`gcloud container clusters get-credentials <your-cluster-name> --region <your-region> --project <your-project>`
with a documentation link: https://cloud.google.com/sdk/gcloud/reference/container/clusters/get-credentials.
We can mention that you can copy the connection command from the GKE console.

We could also mention `kubectl config use-context` as a way of switching to the right cluster, with a link to the docs: https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/

Reply:
Updated.